### PR TITLE
Add v2 dist benchmark vgg

### PR DESCRIPTION
**Pull request by typhoonzero**

### CODE REVIEW

File: benchmark/cluster/vgg16/README.md [hunk 1]
```diff
@@ -0,0 +1,58 @@
...
+
+### different batch size
+
+- PServer Count: 10
+- Trainer Count: 20
+- Metrics: samples / sec
+
+| Batch Size | 32 | 64 | 128 | 256 |
+| -- | -- | -- | -- | -- |
+| PaddlePaddle Fluid | - | 247.40 | - | - |
```
Comment (helinwang) [inline benchmark/cluster/vgg16/README.md]: It seesm fluid's performance is `247.40/64=3.866` batch per second, and v2's performance is `256.14/128=2.001` batch per second.
Seems the different is huge, do you have an idea why? (also please check if my math is correct).

Comment (typhoonzero) [reply benchmark/cluster/vgg16/README.md]: Sorry, wrong column. I'll update this PR with full test result.

File: benchmark/cluster/vgg16/Dockerfile [hunk 2]
```diff
@@ -0,0 +1,15 @@
+#FROM paddlepaddle/paddlecloud-job
+#RUN mkdir -p /workspace
+#ADD reader.py /workspace/
+#RUN python /workspace/reader.py
+FROM python:2.7.14
```
Comment (gongweibao) [inline benchmark/cluster/vgg16/Dockerfile]: 我觉得既然是测试，最好不用这个而是用paddle:dev。
- 不用安装其他的依赖
- 调试的时候进入容器可以用各种命令查看系统的状态。

File: benchmark/cluster/vgg16/Dockerfile [hunk 3]
```diff
@@ -0,0 +1,15 @@
...
+ADD paddle_k8s /usr/bin
+ADD k8s_tools.py /root
+RUN pip install -U kubernetes opencv-python &&   apt-get update -y &&   apt-get install -y iputils-ping libgtk2.0-dev 
+ADD *.whl /
+RUN pip install /*.whl && rm -f /*.whl
+ENV LD_LIBRARY_PATH=/usr/local/lib
+ADD reader.py /workspace/
+RUN python /workspace/reader.py
```
Comment (gongweibao) [inline benchmark/cluster/vgg16/Dockerfile]: 这个基本上下载不下来，所以需要加提示，提示用户使用代理。

File: benchmark/cluster/vgg16/v2_trainer.yaml [hunk 4]
```diff
@@ -0,0 +1,65 @@
...
+        - name: BATCH_SIZE
+          value: "256"
+        - name: TRAINERS
+          value: "20"
+        - name: PSERVERS
+          value: "10"
+        - name: TOPOLOGY
+          value: ""
+        - name: ENTRY
+          value: "cd /workspace && MKL_NUM_THREADS=1 python /workspace/vgg16_v2.py"
```
Comment (gongweibao) [inline benchmark/cluster/vgg16/v2_trainer.yaml]: `python -u`,强制输出日志。

File: benchmark/cluster/vgg16/v2_pserver.yaml [hunk 5]
```diff
@@ -0,0 +1,64 @@
...
+        - name: PADDLE_JOB_NAME
+          value: vgg16v2job
+        - name: TRAINERS
+          value: "20"
+        - name: PSERVERS
+          value: "10"
+        - name: TOPOLOGY
+          value: ""
+        - name: ENTRY
+          value: "python train.py"
```
Comment (gongweibao) [inline benchmark/cluster/vgg16/v2_pserver.yaml]: `python -u`,强制输出日志。

File: benchmark/cluster/vgg16/README.md [hunk 6]
```diff
@@ -0,0 +1,74 @@
...
+- Trainer Count: 20
+- Metrics: samples / sec
+
+| Batch Size | 32 | 64 | 128 | 256 |
+| -- | -- | -- | -- | -- |
+| PaddlePaddle Fluid | 15.44 | 16.32 | 16.74 | 16.79 |
+| PaddlePaddle v2 | 15.97 | 17.04 | 17.60 | 17.83 |
+| TensorFlow | - | - | - | - |
+
+### different batch size
```
Comment (Yancey1989) [inline benchmark/cluster/vgg16/README.md]: > different batch size
=>
> Different Batch Size

File: benchmark/cluster/vgg16/README.md [hunk 7]
```diff
@@ -0,0 +1,74 @@
...
+- Metrics: samples / sec
+
+| Batch Size | 32 | 64 | 128 | 256 |
+| -- | -- | -- | -- | -- |
+| PaddlePaddle Fluid | 190.20 | 222.15 | 247.40 | 258.18 |
+| PaddlePaddle v2 | 170.96 | 233.71 | 256.14 | 329.23 |
+| TensorFlow | - | - | - | - |
+
+
+### Accelerate rate
```
Comment (Yancey1989) [inline benchmark/cluster/vgg16/README.md]: Accelerate Rate

File: benchmark/cluster/vgg16/README.md [hunk 8]
```diff
@@ -0,0 +1,74 @@
...
+- Batch Size: 128
+- Metrics: samples / sec
+
+| Trainer Counter | 20 | 40 | 80 | 100 |
+| -- | -- | -- | -- | -- |
+| PaddlePaddle Fluid | 263.29 | 518.80 | 836.26 | 1019.29 |
+| PaddlePaddle v2 (need more tests) | 326.85 | 534.58 | 853.30 | 1041.99 |
+| TensorFlow | - | - | - | - |
+
+### different pserver number
```
Comment (Yancey1989) [inline benchmark/cluster/vgg16/README.md]: Different PServer Count

File: benchmark/cluster/vgg16/README.md (hunk 7 above)
Comment (Yancey1989) [inline benchmark/cluster/vgg16/README.md]: it's a **rate** metrics, so maybe we need to calculate this value by https://github.com/PaddlePaddle/Paddle/tree/develop/benchmark/cluster#measure-parallel-efficiency-by-increasing-trainer-count ?

File: benchmark/cluster/vgg16/README.md [hunk 9]
```diff
@@ -0,0 +1,76 @@
...
+| -- | -- | -- | -- | -- |
+| PaddlePaddle Fluid | 263.29 (78.64%) | 518.80 (77.47%) | 836.26 (62.44%) | 1019.29 (60.89%) |
+| PaddlePaddle v2 (need more tests) | 326.85 (92.85%) | 534.58 (75.93%) | 853.30 (60.60%) | 1041.99 (59.20%) |
+| TensorFlow | - | - | - | - |
+
+### Different Pserver Count
+
+- Trainer Count: 60
+- Batch Size: 128
+- Metrics: mini-batch / sec
```
Comment (Yancey1989) [inline benchmark/cluster/vgg16/README.md]: > mini-batch / sec

Do you mean `samples / sec` ?

File: benchmark/cluster/vgg16/README.md [hunk 10]
```diff
@@ -0,0 +1,76 @@
...
+1. When the build finishes, copy the output `whl` package located under `build/python/dist` to current directory.
+1. Run `docker build -t [image:tag] .` to build the docker image and run `docker push [image:tag]` to push the image to reponsitory so kubernetes can find it.
+1. Run `kubectl create -f pserver.yaml && kubectl create -f trainer.yaml` to start the job on your kubernetes cluster (you must configure the `kubectl` client before this step).
+1. Run `kubectl get po` to get running pods, and run `kubectl logs [podID]` to fetch the pod log of pservers and trainers.
+
+Check the logs for the distributed training progress and analyze the performance.
+
+## Enable verbos logs
+
+Edit `pserver.yaml` and `trainer.yaml` and add an environment variable `GLOG_v=3` to see what happend in detail.
```
Comment (Yancey1989) [inline benchmark/cluster/vgg16/README.md]: I'm not sure whether we need to add `GLOG_logtostderr=1`, if you have tested it, please ignore this comment.

File: benchmark/cluster/vgg16/Dockerfile [hunk 11]
```diff
@@ -0,0 +1,16 @@
+FROM python:2.7.14
+RUN pip install -U kubernetes opencv-python &&   apt-get update -y &&   apt-get install -y iputils-ping libgtk2.0-dev
+# NOTE: By default CI built wheel packages turn WITH_DISTRIBUTE=OFF,
+#       so we must build one with distribute support to install in this image.
+RUN pip install paddlepaddle
```
Comment (Yancey1989) [inline benchmark/cluster/vgg16/Dockerfile]: Does this `pip install` is redundant? Move the dataset download after line12 ?

Comment (typhoonzero) [reply benchmark/cluster/vgg16/Dockerfile]: No, in order to make debugging faster, lines below changes much, and download dataset is slow, so add this line.

File: paddle/gserver/layers/MultiBoxLossLayer.h [hunk 12]
```diff
@@ -1,3 +1,16 @@
+//  Copyright (c) 2018 PaddlePaddle Authors. All Rights Reserve.
```
Comment (Yancey1989) [inline paddle/gserver/layers/MultiBoxLossLayer.h]: The copyright message is duplicated.

**Pull request merged**